eloquentjs/4_4_deep_comparison.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * Performs deep comparison of two inputs and determines equality
+ * @param {*} itemA 
+ * @param {*} itemB 
+ * @returns 
+ */
+function deepEqual(itemA, itemB) {
+    if(itemA === itemB) {
+        return true;
+    }
+    if (typeof itemA !== typeof itemB) {
+        return false;
+    }
+    if(itemA === null || itemB === null) {
+        return false;
+    }
+    if (typeof itemA !== 'object') {
+        return false;
+    }
+
+    let entriesA = Object.entries(itemA);
+    let entriesB = Object.entries(itemB);
+
+    if (entriesA.length !== entriesB.length) {
+        return false;
+    }
+
+    entriesA.sort();
+    entriesB.sort();
+
+    for (let i = 0; i < entriesA.length; i++) {
+        let keyValuePairA = entriesA[i];
+        let keyValuePairB = entriesB[i];
+
+        let keysEqual = deepEqual(keyValuePairA[0], keyValuePairB[0]);
+        let valuesEqual = deepEqual(keyValuePairA[1], keyValuePairB[1])
+
+        if (!keysEqual) {
+            return false;
+        }
+        if (!valuesEqual) {
+            return false;
+        }
+    }
+
+    return true;
+}
+
+let jimFavorites    = {favoriteNumber: 3, favoriteFruit: 'apple', favoriteDog: {name: 'simon', breed: 'border collie'}, favoriteColors: ['red', 'blue']};
+let janeFavorites   = {favoriteNumber: 3, favoriteFruit: 'apple', favoriteDog: {name: 'simon', breed: 'border collie'}, favoriteColors: ['red', 'blue']}; // Same as jim
```

Comment:
The way these tests are organized is a bit confusing so I'm not sure, but it seems like you're not testing deep equals on objects with different insertion orders. I think you should add some cases like that and try to make the way you've split these cases up clearer.